rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -2221,21 +2234,25 @@ private Integer rollbackFederation(boolean dryRun) {
         FederationConstants federationConstants = bridgeConstants.getFederationConstants();
         provider.getFederationElection(federationConstants.getFederationChangeAuthorizer()).clear();
 
-        return 1;
+        logger.info("[rollbackFederation] Successfully rolled back pending federation.");
+        return FederationChangeResponseCode.SUCCESSFUL.getCode();
     }
 
     public Integer voteFederationChange(Transaction tx, ABICallSpec callSpec) throws BridgeIllegalArgumentException {
+        String calledFunction = callSpec.getFunction();
         // Must be on one of the allowed functions
-        if (!FEDERATION_CHANGE_FUNCTIONS.contains(callSpec.getFunction())) {
-            return FEDERATION_CHANGE_GENERIC_ERROR_CODE;
+        if (!FEDERATION_CHANGE_FUNCTIONS.contains(calledFunction)) {
+            logger.warn("[voteFederationChange] Federation change function {} does not exist.", calledFunction);
+            return FederationChangeResponseCode.NON_EXISTING_FUNCTION_CALLED.getCode();
         }
 
         FederationConstants federationConstants = bridgeConstants.getFederationConstants();
         AddressBasedAuthorizer authorizer = federationConstants.getFederationChangeAuthorizer();
 
         // Must be authorized to vote (checking for signature)
         if (!authorizer.isAuthorized(tx, signatureCache)) {
-            return FEDERATION_CHANGE_GENERIC_ERROR_CODE;
+            logger.warn("[voteFederationChange] Unauthorized voter {}.", tx.getSender(signatureCache));
```

Comment:
```suggestion
            RskAddress voterAddress = tx.getSender(signatureCache);
            logger.warn("[voteFederationChange] Unauthorized voter {}.", voterAddress);
```
I think you need to do it this way or else sonar complains about calling methods only conditionally (or something similar)

Reply:
thanks

Reply:
done!